### PR TITLE
Diagnostics renamed

### PR DESCRIPTION
**Pull request by zhelenskiy**
As inline classes are part of wider functionality value classes diagnostics should be renamed

### CODE REVIEW

File: compiler/fir/checkers/checkers-component-generator/src/org/jetbrains/kotlin/fir/checkers/generator/diagnostics/FirDiagnosticsList.kt [hunk 1]
```diff
@@ -395,25 +395,25 @@ object DIAGNOSTICS_LIST : DiagnosticList("FirErrors") {
...
-        val INLINE_CLASS_CONSTRUCTOR_WRONG_PARAMETERS_SIZE by error<KtElement>()
-        val INLINE_CLASS_CONSTRUCTOR_NOT_FINAL_READ_ONLY_PARAMETER by error<KtParameter>()
-        val PROPERTY_WITH_BACKING_FIELD_INSIDE_INLINE_CLASS by error<KtProperty>(PositioningStrategy.DECLARATION_SIGNATURE)
-        val DELEGATED_PROPERTY_INSIDE_INLINE_CLASS by error<PsiElement>()
-        val INLINE_CLASS_HAS_INAPPLICABLE_PARAMETER_TYPE by error<KtTypeReference> {
+    val VALUE_CLASSES by object : DiagnosticGroup("Value classes") {
+        val VALUE_CLASS_NOT_TOP_LEVEL by error<KtDeclaration>(PositioningStrategy.INLINE_OR_VALUE_MODIFIER)
+        val VALUE_CLASS_NOT_FINAL by error<KtDeclaration>(PositioningStrategy.MODALITY_MODIFIER)
+        val ABSENCE_OF_PRIMARY_CONSTRUCTOR_FOR_VALUE_CLASS by error<KtDeclaration>(PositioningStrategy.INLINE_OR_VALUE_MODIFIER)
+        val VALUE_CLASS_CONSTRUCTOR_WRONG_PARAMETERS_SIZE by error<KtElement>()
```
Comment (ilmirus) [inline compiler/fir/checkers/checkers-component-generator/src/org/jetbrains/kotlin/fir/checkers/generator/diagnostics/FirDiagnosticsList.kt:402]: This diagnostic should remain for inline classes.

File: compiler/frontend/src/org/jetbrains/kotlin/diagnostics/rendering/DefaultErrorMessages.java [hunk 2]
```diff
@@ -742,20 +742,20 @@ public static DiagnosticRenderer getRendererForDiagnostic(@NotNull UnboundDiagno
...
-        MAP.put(INLINE_CLASS_CANNOT_IMPLEMENT_INTERFACE_BY_DELEGATION, "Inline class cannot implement an interface by delegation");
-        MAP.put(INLINE_CLASS_CANNOT_EXTEND_CLASSES, "Inline class cannot extend classes");
-        MAP.put(INLINE_CLASS_CANNOT_BE_RECURSIVE, "Inline class cannot be recursive");
-        MAP.put(RESERVED_MEMBER_INSIDE_INLINE_CLASS, "Member with the name ''{0}'' is reserved for future releases", STRING);
-        MAP.put(SECONDARY_CONSTRUCTOR_WITH_BODY_INSIDE_INLINE_CLASS, "Secondary constructors with bodies are reserved for for future releases");
-        MAP.put(INNER_CLASS_INSIDE_INLINE_CLASS, "Inline class cannot have inner classes");
+        MAP.put(VALUE_CLASS_NOT_TOP_LEVEL, "Value classes cannot be local or inner");
+        MAP.put(VALUE_CLASS_NOT_FINAL, "Value classes can be only final");
+        MAP.put(ABSENCE_OF_PRIMARY_CONSTRUCTOR_FOR_VALUE_CLASS, "Primary constructor is required for value class");
+        MAP.put(VALUE_CLASS_CONSTRUCTOR_WRONG_PARAMETERS_SIZE, "Inline class must have exactly one primary constructor parameter"); // +
```
Comment (ilmirus) [inline compiler/frontend/src/org/jetbrains/kotlin/diagnostics/rendering/DefaultErrorMessages.java:748]: Meaningless comment

File: compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/diagnostics/FirErrorsDefaultMessages.kt [hunk 3]
```diff
@@ -1051,43 +1052,44 @@ object FirErrorsDefaultMessages : BaseDiagnosticRendererFactory() {
...
-        // Inline and value classes
-        map.put(INLINE_CLASS_NOT_TOP_LEVEL, "Inline classes cannot be local or inner")
-        map.put(INLINE_CLASS_NOT_FINAL, "Inline classes can be only final")
-        map.put(ABSENCE_OF_PRIMARY_CONSTRUCTOR_FOR_INLINE_CLASS, "Primary constructor is required for inline class")
-        map.put(INLINE_CLASS_CONSTRUCTOR_WRONG_PARAMETERS_SIZE, "Inline class must have exactly one primary constructor parameter")
+        // Value classes
+        map.put(VALUE_CLASS_NOT_TOP_LEVEL, "Value classes cannot be local or inner")
+        map.put(VALUE_CLASS_NOT_FINAL, "Value classes can be only final")
+        map.put(ABSENCE_OF_PRIMARY_CONSTRUCTOR_FOR_VALUE_CLASS, "Primary constructor is required for value class")
+        map.put(VALUE_CLASS_CONSTRUCTOR_WRONG_PARAMETERS_SIZE, "Inline class must have exactly one primary constructor parameter") // +
```
Comment (ilmirus) [inline compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/diagnostics/FirErrorsDefaultMessages.kt:1059]: Meaningless comment

**Review by ilmirus — CHANGES_REQUESTED** [review]

File: compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/diagnostics/FirErrorsDefaultMessages.kt (hunk 3 above)
Comment (zhelenskiy) [reply compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/diagnostics/FirErrorsDefaultMessages.kt:1059]: According to comment in the beginning of the file this means that a better message is expected. In this case I set it as value classes may have more than 1 argument but it is not yet supported.

File: compiler/frontend/src/org/jetbrains/kotlin/diagnostics/rendering/DefaultErrorMessages.java (hunk 2 above)
Comment (zhelenskiy) [reply compiler/frontend/src/org/jetbrains/kotlin/diagnostics/rendering/DefaultErrorMessages.java:748]: I put that comment for the same reasons as in Fir file.

**Review by ilmirus — APPROVED** [review]

**Comment by ilmirus** [comment]
Thanks! I manually pushed the changes to master.

**Pull request closed without merging**